docs/source/frameworks/onnx.rst

```diff
@@ -0,0 +1,460 @@
+====
+ONNX
+====
+
+
+Preface
+-------
+
+ONNX is an open format built to represent machine learning models. ONNX provides `high interoperability <https://onnx.ai/supported-tools.html#buildModel>`_  among various frameworks, as well as enable machine learning practitioners to maximize models' performance across `different hardware <https://onnx.ai/supported-tools.html#deployModel>`_.
+
+Due to its high interoperability among frameworks, we recommend you to check out the framework integration with ONNX as it will contain specific recommendation and requirements for that given framework.
+
+
+.. tab:: PyTorch
+
+   - `Quick tutorial about exporting a model from PyTorch to ONNX <https://pytorch.org/tutorials/advanced/super_resolution_with_onnxruntime.html>`_ from official PyTorch documentation.
+   - `torch.onnx <https://pytorch.org/docs/stable/onnx.html>`_ from official PyTorch documentation. Pay special attention to section **Avoiding Pitfalls**, **Limitations** and **Frequently Asked Questions**.
+
+.. tab:: TensorFlow
+
+   - `tensorflow-onnx (tf2onnx) <https://github.com/onnx/tensorflow-onnx>`_ documentation.
+
+.. tab:: Scikit Learn
+
+   TODO
+
+Converting model frameworks to ONNX format
+-----------------------------------------------
+
+.. note::
+
+   BentoML currently only support `ONNX Runtime
+   <https://onnxruntime.ai>`_ as ONNX backend. BentoML requires either
+   ``onnxruntime>=1.9`` or ``onnxruntime-gpu>=1.9`` to be installed.
+
+.. tab:: PyTorch
+
+   First, let’s create a SuperResolution model in PyTorch.
+
+   .. code-block:: python
+
+      import torch.nn as nn
+      import torch.nn.init as init
+
+      class SuperResolutionNet(nn.Module):
+	  def __init__(self, upscale_factor, inplace=False):
+	      super(SuperResolutionNet, self).__init__()
+
+	      self.relu = nn.ReLU(inplace=inplace)
+	      self.conv1 = nn.Conv2d(1, 64, (5, 5), (1, 1), (2, 2))
+	      self.conv2 = nn.Conv2d(64, 64, (3, 3), (1, 1), (1, 1))
+	      self.conv3 = nn.Conv2d(64, 32, (3, 3), (1, 1), (1, 1))
+	      self.conv4 = nn.Conv2d(32, upscale_factor ** 2, (3, 3), (1, 1), (1, 1))
+	      self.pixel_shuffle = nn.PixelShuffle(upscale_factor)
+
+	      self._initialize_weights()
+
+	  def forward(self, x):
+	      x = self.relu(self.conv1(x))
+	      x = self.relu(self.conv2(x))
+	      x = self.relu(self.conv3(x))
+	      x = self.pixel_shuffle(self.conv4(x))
+	      return x
+
+	  def _initialize_weights(self):
+	      init.orthogonal_(self.conv1.weight, init.calculate_gain('relu'))
+	      init.orthogonal_(self.conv2.weight, init.calculate_gain('relu'))
+	      init.orthogonal_(self.conv3.weight, init.calculate_gain('relu'))
+	      init.orthogonal_(self.conv4.weight)
+
+      torch_model = SuperResolutionNet()
+
+   For this tutorial, we will download some pre-trained weights. Note
+   that this model was not trained fully for good accuracy and is used
+   here for demonstration purposes only.
+
+   .. code-block:: python
```

Comment:
```suggestion
```

No need to explain about this here. We maybe can just link to the super resolution tutorial. It should also use a pretrained model right?

Reply:
I think these sentences is for user who really followed our guides and then find out the result is suboptimal, then they will know it's not their faults (bad copy and paste) or our faults. @aarnphm